script/tool/lib/src/build_examples_command.dart

```diff
@@ -99,38 +99,60 @@ class BuildExamplesCommand extends PackageLoopingCommand {
         .map((MapEntry<String, _PlatformDetails> entry) => entry.value);
-    final Set<_PlatformDetails> buildPlatforms = <_PlatformDetails>{};
-    final Set<_PlatformDetails> unsupportedPlatforms = <_PlatformDetails>{};
-    for (final _PlatformDetails platform in requestedPlatforms) {
-      if (pluginSupportsPlatform(platform.pluginPlatform, package)) {
-        buildPlatforms.add(platform);
-      } else {
-        unsupportedPlatforms.add(platform);
-      }
-    }
+
+    // Platform support is checked at the package level for plugins; there is
+    // no package-level platform information for non-plugin packages.
+    final Set<_PlatformDetails> buildPlatforms = isPlugin
+        ? requestedPlatforms
+            .where((_PlatformDetails platform) =>
+                pluginSupportsPlatform(platform.pluginPlatform, package))
+            .toSet()
+        : requestedPlatforms.toSet();
     if (buildPlatforms.isEmpty) {
       final String unsupported = requestedPlatforms.length == 1
           ? '${requestedPlatforms.first.label} is not supported'
-          : 'None of [${requestedPlatforms.map((_PlatformDetails p) => p.label).join(',')}] are supported';
+          : 'None of [${requestedPlatforms.map((_PlatformDetails p) => p.label).join(', ')}] are supported';
       return PackageResult.skip('$unsupported by this plugin');
     }
     print('Building for: '
-        '${buildPlatforms.map((_PlatformDetails platform) => platform.label).join(',')}');
+        '${buildPlatforms.map((_PlatformDetails platform) => platform.label).join(', ')}');
+
+    final Set<_PlatformDetails> unsupportedPlatforms =
+        requestedPlatforms.toSet().difference(buildPlatforms);
     if (unsupportedPlatforms.isNotEmpty) {
+      final List<String> skippedPlatforms = unsupportedPlatforms
+          .map((_PlatformDetails platform) => platform.label)
+          .toList();
+      skippedPlatforms.sort();
       print('Skipping unsupported platform(s): '
-          '${unsupportedPlatforms.map((_PlatformDetails platform) => platform.label).join(',')}');
+          '${skippedPlatforms.join(', ')}');
     }
     print('');
 
-    for (final Directory example in getExamplesForPlugin(package)) {
+    bool builtSomething = false;
+    for (final Directory example in getExamplesForPackage(package)) {
       final String packageName =
           getRelativePosixPath(example, from: packagesDir);
 
       for (final _PlatformDetails platform in buildPlatforms) {
+        // Repo policy is that a plugin must have examples configured for all
+        // supported platforms. For packages, just log and skip any requested
+        // platform that a package doesn't have set up.
+        if (!isPlugin &&
+            !example
+                .childDirectory(platform.flutterPlatformDirectory)
+                .existsSync()) {
+          print('Skipping ${platform.label} for $packageName; not supported.');
+          continue;
```

Comment:
nit: Instead of using `continue` do you think an `else` clause would be preferable?

Reply:
I generally don't like to nest non-trivial primary logic in an else clause.

Reply:
SGTM, not my preference but not in violation of our style.  Academics have argued about it for decades: https://en.wikipedia.org/wiki/Structured_programming#Early_exit